### ISSUE TITLE
df_print option is ignored in Distill YAML header

### ISSUE THREAD

**Issue by Selbosh** [rstudio/distill#310]
Right now, any attempt to set the `df_print` option in YAML is ignored when rendering articles or posts.

As far as I can tell, this option is not [passed to `render_context()`](https://github.com/rstudio/rmarkdown/blob/1707d0b5189e3f12caf16a6cd53391f5be2b2f08/R/knit_print.R) so R Markdown isn't able to pick it up when knitting the document. This is very annoying and seems inconsistent with almost every other R Markdown output format out there.

-------------

### Workaround

Current workaround is as follows: add a chunk near the top of your Rmd file with the following, assuming you want `df_print: kable`.

```r
library(knitr)
# define a method for objects of the class data.frame
knit_print.data.frame = function(x, ...) {
  res = paste(c('', '', kable(x, 'html'), ''), collapse = '\n')
  asis_output(res)
}
# register the method
registerS3method("knit_print", "data.frame", knit_print.data.frame)
```

Then your Distill articles will knit as if they were another format with `df_print: kable` in the header.

**Comment by cderv** [comment]
Yes currently `df_print` is not an option in `distill_article()` which is based on `rmarkdown::html_document_base`, so don't inherit the argument one from `rmarkdown::html_document`. 

Tables format has to be provided at the chunk level: https://rstudio.github.io/distill/tables.html

@jjallaire is there anything against adding a `df_print:` feature to **distill** ?

**Comment by Selbosh** [comment]
Anyone needing a workaround for `df_print: paged`, it's a similar solution to above.

```r
library(knitr)
knit_print.data.frame <- function(x, ...) {
  asis_output(
    rmarkdown:::paged_table_html(x, options = attr(x, "options")),
    meta = list(dependencies = rmarkdown:::html_dependency_pagedtable())
  )
}
registerS3method("knit_print", "data.frame", knit_print.data.frame)
```

Based on https://github.com/rstudio/rmarkdown/blob/61db7a9f5b18d77609877e3a3a039758b17d554a/R/html_paged.R#L241
